zcash_client_sqlite/src/wallet/init/migrations/addresses_table.rs

```diff
@@ -64,9 +62,8 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
 
```

Comment:
Pre-existing issue: because this uses multiple transactions, a failure at the wrong time will get the database into a state where the `addresses` and `accounts_new` exist, but the migration has not completed. Because `CREATE TABLE` (rather than `CREATE TABLE IF NOT EXISTS`) is used, this may (the documentation isn't entirely clear) [cause an error](https://www.sqlite.org/lang_createtable.html):

> It is usually an error to attempt to create a new table in a database that already contains a table, index or view of the same name. However, if the "IF NOT EXISTS" clause is specified as part of the CREATE TABLE statement and a table or view of the same name already exists, the CREATE TABLE command simply has no effect (and no error message is returned). An error is still returned if the table cannot be created because of an existing index, even if the "IF NOT EXISTS" clause is specified.

This would brick the wallet, I think.

Reply:
@daira the entire migration takes place inside a single transaction.

Reply:
This doesn't use multiple transactions? The migration is given `&rusqlite::Transaction`, so every individual action inside the `RusqliteMigration::up` impl is performed in a single transaction. The `execute_batch` is just a convenience call for running multiple SQL commands in one Rust call; it doesn't create a separate transaction.